scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,253 @@
+        # Restyle the source files to the new style
+        if code_style.fix_style(changed_src_files) != 0:
+            return False
+
+    # Add the changed files
+    for changed_file in changed_files:
+        result = subprocess.run(["git", "add", changed_file], check=False)
+        # Ignore failures as these are caused by renamed files
+
+    # Commit the newly added files
+    result = subprocess.run(["git", "commit", "--reuse-message=" + \
+            commit_hash], check=False)
+    if result.returncode != 0:
+        print("Error making restyled commit.", file=sys.stderr)
+        return False
+
+    # Cleanup leftover unstaged files
+    result = subprocess.run(["git", "restore", "."], check=False)
+    if result.returncode != 0:
+        print("Error cleaning up unstaged files.", file=sys.stderr)
+        return False
+
+    return True
+
+def list_revisions(revision_or_range: str) -> List[str]:
+    """
+    Return the list of commits in revision_or_range.
+
+    If revision_or_range is a single revision, return it in a one-element
+    list. Otherwise return the list of commits in that range.
+
+    Return the list in order from oldest to newest.
+    """
+    result = subprocess.run(["git", "rev-list", "--no-walk", \
+            revision_or_range], stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting revision list for '"+revision_or_range+"'.", \
+                file=sys.stderr)
+        return []
+
+    return list(reversed(str(result.stdout, "ascii").split()))
+
+def rewrite_branch_code_style(existing_branch: str, new_branch: str, \
+        codestyle_change_commit: str) -> bool:
+    """
+    Rewrite the given existing branch onto the given new branch on top of the
+    given commit that changes the code style.
+    """
+    # Create the new branch to rewrite onto at the code style change commit
+    result = subprocess.run(["git", "checkout", "--detach",
+                             codestyle_change_commit], \
+            check=False)
+    if result.returncode != 0:
+        print("Error checking out code style commit '" + \
+                codestyle_change_commit + "'.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "checkout", "-b", new_branch], check=False)
+    if result.returncode != 0:
+        print("Error creating destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # List all the commits between the commit before the codestyle change
+    # (where the branch diverged) and the tip of the branch.
+    for rev in list_revisions(codestyle_change_commit+"~.."+existing_branch):
+        if not restyle_commit_onto_current(rev):
+            delete_branch(new_branch)
+            return False
+
+    return True
+
+def main() -> int:
+    """
+    Main with command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('existing_branch', \
```

Comment:
Positional arguments are perfectly fine. If argparse ever had trouble with them, that was a long time ago and we don't care about such antique versions. (Also, positional arguments are such a basic features that this must have been in the 2.x era before people moved over from `optparse`.) [Python 3.2](https://docs.python.org/3.2/library/argparse.html) already supports `nargs='*'` and we don't need this new script to support Python versions that are out of support (so [requiring 3.7 would be fine](https://devguide.python.org/versions/), although if this must be able to run on our CI we currently need to maintain compatibility with 3.5 because we're behind the times).